core/domain/app_feedback_report_services.py

```diff
@@ -0,0 +1,86 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services to operate on app feedback report models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+
+(app_feedback_report_models,) = models.Registry.import_models(
+    [models.NAMES.app_feedback_report])
+transaction_services = models.Registry.import_transaction_services()
+
+
+def scrub_report(report_id, scrubbed_by):
+    """Scrubs the instance of AppFeedbackReportModel with given ID, removing
+    any user-entered input in the entity.
+
+    Args:
+        report_id: str. The id of the model entity to scrub.
+        scrubbed_by: str. The id of the user or cron job that is intiating
+            scrubbing this report.
+    """
+    _scrub_report_in_transaction(report_id, scrubbed_by)
+
+
+@transaction_services.run_in_transaction_wrapper
+def _scrub_report_in_transaction(report_id, scrubbed_by):
+    """See scrub_report for general documentation of what this method does.
+    It's only safe to call this method from within a transaction.
+
+    Args:
+        report_id: str. The id of the model entity to scrub.
+        scrubbed_by: str. The id of the user that is initiating scrubbing of
+            this report, or a const feconf.REPORT_SCRUBBER_BOT_ID if scrubbed
+            by the cron job.
+    """
+    report_entity = app_feedback_report_models.AppFeedbackReportModel.get_by_id(
+        report_id)
+    if not report_entity:
+        raise Exception(
+            'The AppFeedbackReportModel trying to be scrubbed does not '
+            'exist.')
+    if report_entity.platform == (
+            app_feedback_report_models.PLATFORM_CHOICE_ANDROID):
+        scrubbed_report_info = _scrub_report_info(
+            report_entity.android_report_info)
+        report_entity.android_report_info = scrubbed_report_info
+    else:
+        scrubbed_report_info = _scrub_report_info(
+            report_entity.web_report_info)
+        report_entity.web_report_info = scrubbed_report_info
+    report_entity.scrubbed_by = scrubbed_by
+    report_entity.update_timestamps()
+    report_entity.put()
+
+
+def _scrub_report_info(report_info_dict):
+    """Scrubs the dictionary of any fields that contains input directly from
+    the user.
+
+    Args:
+        report_info_dict: dict. The info dict collected in a report.
+
+    Returns:
+        dict. The scrubbed report.
+    """
+    new_report_info = dict()
+    for key in report_info_dict:
+        if key not in app_feedback_report_models.REPORT_INFO_TO_REDACT:
+            new_report_info[key] = report_info_dict[key]
```

Comment:
Dicts should have well-defined schemas, and using domain objects (see above comment) will help enforce that. Note that you will need sub-domain-objects for the sub-models. The currently-implemented approach does not work well because the schemas would differ between scrubbed and non-scrubbed dicts.

Btw, you might want to make these changes in a separate PR that focuses more on the domain layer, to avoid this one getting too large.

Reply:
Ack, is it ok to make the above change and [this other change](https://github.com/oppia/oppia/pull/12298#discussion_r604720415) in a later PR? I've started the domain objects in a separate PR already. This then means that the [next PR](https://github.com/oppia/oppia/pull/12367) for the scrubbing job will still use this current implementation that manipulates the model

Reply:
No, could you remove these functions from the current PR instead? We shouldn't commit incorrect stuff.

Reply:
Done, removed the services and tests. I think what might have to happen then is that the order of PRs will need to be storage model > domain objects > scrubber, as an FYI.